pkg/hubble/parser/getters/getters.go

```diff
@@ -27,7 +27,8 @@ import (
 // DNSGetter ...
 type DNSGetter interface {
 	// GetNamesOf fetches FQDNs of a given IP from the perspective of
-	// the endpoint with ID sourceEpID
+	// the endpoint with ID sourceEpID. The returned names must not have
+	// trailing dots.
```

Comment:
even though the reason is included in the commit, commits are volatile. I think the same reason should be included here since devs are more likely to find this comment than the commit message.

Reply:
on a side note: if this is a requirement, shouldn't a consumer of DNSGetter strip the dot so there is no new getters added without this requirement?

Reply:
I can extend the comment in a follow-up PR. I would like to avoid to restart CI just for this.

>  on a side note: if this is a requirement, shouldn't a consumer of DNSGetter strip the dot so there is no new getters added without this requirement?

It's part of the interface implementation contract to not return trailing dots. Having to modify each and every call site to clean up the result because it was validating the implementation contract seems weird to me.